src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java

```diff
@@ -134,6 +144,57 @@ private static Token retrieveTokenWithPasswordGrant(Properties properties, OkHtt
         }
     }
 
+    private static Token retrieveTokenWithJWTBearerGrant(Properties properties, OkHttpClient client) throws TokenException {
+        if (properties.getProperty(Constants.USER_NAME) == null || properties.getProperty(Constants.USER_NAME).isEmpty()) {
```

Comment:
I think this case is handled above in getCondition as well. Can we handle this in one place

Reply:
Kept this check incase function 'retrieveTokenWithJWTBearerGrant' is used at different places in the future.